heckin-case-converter.qmd

```diff
@@ -0,0 +1,313 @@
+---
+title: "A package from start to finish"
+subtitle: "making a heckin' case converter"
+freeze: true
+---
+
+The Rust crate ecosystem is rich with very small and very powerful utility libraries. One of the most downloaded crates is [heck](https://docs.rs/heck). It provides traits and structs to perform some of the most common case conversions.
+
+In this tutorial we'll create a 0 dependency R package to provide the common case conversions. The resultant R package will be more performant but less flexible than the [`{snakecase}`](https://tazinho.github.io/snakecase/) R package. 
+
+This tutorial covers: 
+
+- vectorization
+- `NA` handling
+- code generation using a macro
+
+## Getting started
+
+Create a new R package:
+
+```r
+usethis::create_package("heck")
+```
+
+When the new R package has opened up, add `extendr`.
+
+```r
+rextendr::use_extendr(crate_name = "rheck", lib_name = "rheck")
+```
+
+::: callout-note
+When adding the extendr dependency, make sure that the `crate_name` and `lib_name` arguments _are not_ `heck`. In order to add the `heck` crate as a dependency, the crate itself cannot be called `heck` because it creates a recursive dependency. Doing this allows us to name the R package `{heck}`, but the internal Rust crate is called `rheck`.
```

Comment:
_technically_ https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#renaming-dependencies-in-cargotoml

Reply:
Do you want to adjust the language here? Or we could add a collapsed callout block and "akshully"

Reply:
lol okay whew